reasoner/resolution/resolver/SubsumptiveCoordinator.java

```diff
@@ -63,14 +73,28 @@ public void receiveRevisit(Request.Revisit fromUpstream) {
         LOG.trace("{}: received Revisit: {}", name(), fromUpstream);
         assert isInitialised;
         if (isTerminated()) return;
-        Driver<? extends Resolver<?>> worker = getOrCreateBoundResolver(fromUpstream.visit().partialAnswer());
-        Request.Factory requestFactory = getOrCreateRequestFactory(fromUpstream.visit().partialAnswer(), worker);
+        ConceptMap conceptMap = fromUpstream.visit().partialAnswer().conceptMap();
+        Mapping mapping = reflexiveMappings.get(conceptMap);
+        if (mapping == null) mapping = computeReflexiveMappings(conceptMap);
+        Driver<? extends Resolver<?>> worker = getOrCreateBoundResolver(fromUpstream.visit().partialAnswer(),
+                                                                        mapping.transform(conceptMap));
+        Request.Factory requestFactory = getOrCreateRequestFactory(fromUpstream.visit().partialAnswer(), worker, mapping);
         Request.Revisit revisit = requestFactory.createRevisit(fromUpstream.trace(), fromUpstream.cycles());
         revisitDownstream(revisit, fromUpstream);
     }
 
-    private Request.Factory getOrCreateRequestFactory(AnswerState.Partial<?> partial, Driver<? extends Resolver<?>> receiver) {
-        return requestFactories.computeIfAbsent(partial, p -> Request.Factory.create(driver(), receiver, p));
+    protected Mapping computeReflexiveMappings(ConceptMap conceptMap) {
+        equivalentMappings.forEach(m -> reflexiveMappings.put(m.unTransform(conceptMap), m));
+        return reflexiveMappings.get(conceptMap);
+    }
+
+    private Request.Factory getOrCreateRequestFactory(AnswerState.Partial<?> partial, Driver<? extends Resolver<?>> receiver, Mapping mapping) {
+        return requestFactories.computeIfAbsent(partial, p -> Request.Factory.create(driver(), receiver, applyRemapping(p, mapping)));
+    }
```

Comment:
note to future iterations: this is using a lot of memory - trade off here is a model strangeness of creating a factory each time vs saving the factory. Hopefully this goes away in the new model since there is no clean/good solution here.